core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -362,7 +401,7 @@ class LogManager(val logDirs: Array[File],
         val dataDir = nextLogDir()
         val dir = new File(dataDir, topicPartition.topic + "-" + topicPartition.partition)
         dir.mkdirs()
-        val log = new Log(dir, config, recoveryPoint = 0L, scheduler, time)
+        val log = new Log(dir, config, logStartOffset = 0L, recoveryPoint = 0L, scheduler, time)
```

Comment:
We don't read the logStartOffset from the checkpoint in this case, which is a bit inconsistent with how we load the log when the broker starts up. This seems ok for now since in the rare case when a follower loses the whole log (but still has the logStartOffset checkpoint), it can discover the logStartOffset from the leader quickly. Not sure if we want to add a comment on this.